pkg/instance/kubernetes.go

```diff
@@ -141,32 +141,43 @@ func (ks kubernetesService) getLogs(instance *model.Instance, selector string) (
 		Stream(context.TODO())
 }
 
-func (ks kubernetesService) getPod(instance *model.Instance, selector string) (v1.Pod, error) {
-	var labelSelector string
-	if selector == "" {
-		labelSelector = fmt.Sprintf("im-id=%d", instance.ID)
+func (ks kubernetesService) getPod(instance *model.Instance, typeSelector string) (v1.Pod, error) {
+	labels := &metav1.LabelSelector{
+		MatchLabels: map[string]string{
+			"im-id": fmt.Sprint(instance.ID),
+		},
+	}
+	if typeSelector == "" {
+		labels.MatchLabels["im-default"] = "true"
 	} else {
-		labelSelector = fmt.Sprintf("im-%s-id=%d", selector, instance.ID)
+		labels.MatchLabels["im-type"] = typeSelector
 	}
 
+	labelSelector, err := metav1.LabelSelectorAsSelector(labels)
+	if err != nil {
+		return v1.Pod{}, err
+	}
 	listOptions := metav1.ListOptions{
-		LabelSelector: labelSelector,
+		LabelSelector: labelSelector.String(),
 	}
 
 	podList, err := ks.client.CoreV1().Pods("").List(context.TODO(), listOptions)
```

Comment:
s/podList/pods/ ? I'm well aware the variable was already there ;-)

Reply:
sure